src/main/java/io/gingersnapproject/runtime/BackendListener.java

```diff
@@ -0,0 +1,46 @@
+package io.gingersnapproject.runtime;
+
+import java.net.URI;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+
+import io.gingersnapproject.cdc.event.BackendFailedEvent;
+import io.gingersnapproject.cdc.event.BackendStartedEvent;
+import io.gingersnapproject.cdc.event.BackendStoppedEvent;
+
+import org.eclipse.microprofile.health.Liveness;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Liveness
+@ApplicationScoped
+public class BackendListener extends AbstractHealthChecker<URI> {
```

Comment:
Instead of the generic name "Listener", I would call this the type of the Health check, e.g. "BackendLiveness" and/or "BackendReadiness".